File: codeforces_round_1056/abraham_great_escape.py
# https://codeforces.com/contest/2155/problem/B

t = int(input())
t+=1

while t:=t-1:
    n, k = map(int,input().split())
    if k == n*n-1:
        print('NO')
        continue
    print('Yes')
    for i in range(n):
        for j in range(n):
            is_k_left = k > 0
            is_last_row = i == n-1
            is_last_column = j == n-1

            if is_k_left:
                print('U',end='')
                k -= 1
                continue
            if not is_last_row:
                print('D',end='')
                continue
            if not is_last_column:
                print('R',end='')
                continue
            print('L',end='')
            

        print()

File: codeforces_round_1056/abraham_great_escape.cpp
// https://codeforces.com/contest/2155/problem/B
#include <bits/stdc++.h>
using namespace std;

int main(){
    int t;
    cin >> t;
    while (t--) {
        int n, k;
        cin >> n >> k;

        if (k == n*n-1){
            cout << "NO" << "\n";
        } else {
            cout << "YES" << "\n";
            
            for (int i = 0; i < n; i++){
                for (int j = 0; j < n; j++){

                    if (k-- > 0){
                        cout << "U";
                    } else {
                        if (i != n-1){ // not in last line
                            cout << "D";
                        }
                        else if (j != n-1){ // not in last column
                            cout << "R";
                        } else {
                            cout << "L"; // last element always L
                        }
                    }
                }
                cout << "\n";
            }

        }
    }

}